### ISSUE TITLE
AV on AMP pages fails when going direct to `.amp`

### ISSUE THREAD

**Issue by pjlee11** [bbc/simorgh#3970]
**Note: AMP tests have been disabled due to this bug causing the cypress tests to fail**

These PRs should be reverted as part of closing this issue:

- https://github.com/bbc/simorgh/pull/4591
- https://github.com/bbc/simorgh/pull/4599

**Describe the bug**
AV content on TEST for AMP page fails based on:
```
Origin of <amp-iframe> must not be equal to container  
<amp-iframe 
  sandbox=​"allow-scripts allow-same-origin" 
  layout=​"fill" frameborder=​"0" 
  src=​"/​ws/​av-embeds/​media/​bbc_korean_radio/​liveradio" 
  allowfullscreen=​"true" 
  class=​"i-amphtml-element i-amphtml-layout-fill i-amphtml-layout-size-defined i-amphtml-error i-amphtml-layout" 
  i-amphtml-layout=​"fill" 
  style=​"--loader-delay-offset:​48ms !important;​">
    ​…
​</amp-iframe>​ 
if allow-same-origin is set. See https://github.com/ampproject/amphtml/blob/master/spec/amp-iframe-origin-policy.md for details.
```

**Please note AV playback works when the page is served from AMP cache as the CDN domain is a different domain to `bbc.com`. This only fails when hitting `www.bbc.com/{service}/{assetId}.amp` directly which isn't a real user journey for the audience.**

**To Reproduce**

1. Go to https://www.test.bbc.com/korean/bbc_korean_radio/liveradio.amp
2. Open the console
3. See error

1. Go to https://www.test.bbc.co.uk/news/articles/c3wmq4d1y3wo.amp
2. Open the console
3. See error

**Expected behavior**
The AV content on TEST should have valid playback and no errors should appearing in the console. 

**Screenshots**
![image](https://user-images.githubusercontent.com/7791726/65610921-f8507400-dfa9-11e9-86d1-c0d371ee8dcb.png)
![image](https://user-images.githubusercontent.com/7791726/65610933-fe465500-dfa9-11e9-88f7-217fc207f10a.png)

**Testing notes**
Once fix is promoted to TEST ensure AV content plays back on AMP pages.

- [x] This bug fix is expected to need manual testing.

**Additional context**
https://github.com/ampproject/amphtml/blob/master/spec/amp-iframe-origin-policy.md

**Comment by pjlee11** [comment]
We are currently doing something that is "strongly discouraged" - [here](https://github.com/bbc/psammead/blob/3da6a4fefc43f513a899a4c06e86f853282f4fc2/packages/components/psammead-media-player/src/Amp/index.jsx#L20)

![image](https://user-images.githubusercontent.com/7791726/65615506-5f255b80-dfb1-11e9-9487-fdbc4c27b248.png)

**Comment by pjlee11** [comment]
But then again the docs give it as an [example](https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/amp-iframe.md)

![image](https://user-images.githubusercontent.com/7791726/65616029-3d78a400-dfb2-11e9-926a-b3472ac7fc19.png)

**Comment by pjlee11** [comment]
It looks like alephAMP works based on `www.test.bbc.com` hosted AMP pages requesting the AV embed from `www.bbc.com` 

![image](https://user-images.githubusercontent.com/7791726/65617328-70bc3280-dfb4-11e9-8d64-fd7a5cd048f8.png)

Then on the LIVE env it requests on `www.bbc.com` which as it matches the same domain then AV playback fails

![image](https://user-images.githubusercontent.com/7791726/65617817-44ed7c80-dfb5-11e9-8e5c-4fe9952a6960.png)

**Comment by pjlee11** [comment]
VJ includes work because they come from `news.files.` 

![image](https://user-images.githubusercontent.com/7791726/65621746-3d7da180-dfbc-11e9-8304-635f73eb0498.png)

**Comment by pjlee11** [comment]
To summarise all the above information...

For the SMP instance in the `<amp-iframe>` to be able to play AV we need to set the attribute `sandbox="allow-scripts allow-same-origin"` because SMP needs to run JS (`allow-scripts`) and it needs to download the playlist and access cookies (`allow-same-origin`). 

>An amp-iframe must not be in the same origin as the container unless they do not allow allow-same-origin in the sandbox attribute. See the "Iframe origin policy" doc for further details on allowed origins for iframes.
_[Source](https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/amp-iframe.md#behavior)_

For TEST we are currently requesting the iframe from the same domain therefore AMP is stopping it from rendering. Therefore we need to host the AV embeds on a different domain to the page. Ideally we should have a standalone AV embed domain for this purpose, but if that takes a long time to setup we could do work to request the embed from `.co.uk` when on `.com` and vice versa as a temporary solution. 

**TL;DR**
We need to request the AV embed from a different domain than the one we are serving on. The ideal fix is to have a standalone domain for AV embeds. A hacky temporary fix would be to have `.com` traffic request from `.co.uk` and vice versa with an ADR to explain it.

**Comment by pjlee11** [comment]
Also worth mentioning that Twitter links directly to your AMP page if it finds the `<link rel="amphtml">` and it is valid AMP. This would mean that potentially in production, where alephAMP is seeing the same problems described in this ticket, that any user coming from Twitter to an AMP page might see all AV failing at current.

>Twitter supports the standard AMP discovery mechanism. When a link is shared on Twitter, the Twitter crawler (user agent:Twitterbot/1.0) will look for a <link rel="amphtml"> element in your document. If a valid AMP document is referenced, Twitter mobile clients will direct users to that AMP edition of your page.
_[Source](https://developer.twitter.com/en/docs/publisher-tools/amp/overview)_

**Comment by 12** [comment]
Another good explanation as to why this is the case: https://github.com/ampproject/amphtml/issues/17212#issuecomment-409241528

I think we should serve AV on a separate domain. I am hesitant to just switching TLDs in the meantime, how much extra work would it take to get a separate domain set up? 

Would it not make sense to just go ahead and get this extra domain setup than spend time writing this TLD switching mechanism only to take it out before AV gets enabled on live?

**Comment by graeme-lindsay** [comment]
Pending decisions, so moving to blocked. @dgm34 has been contacted.

**Comment by dgm34** [comment]
This was mulled over by @matthew1000 and @bigmartyn recently when the Forge endpoint was being replaced.

The decision was to use `polling.api.bbci.co.uk`.

This domain is described as being for:
```This is high traffic, high frequency requests that are mostly non-personalised. They can update frequently and so have short max-age values.```

Search for _"Domains used by Web Core"_ on Confluence for additional context.

Some people have suggested purchasing another domain for this purpose. While I am not enthusiastic about this prospect, we could consider renaming `polling.api.bbci.co.uk` if there are better suggestions: it is not yet live and will require an OTG ticket to get set up.

**Comment by chris-hinds** [comment]
Why do we need to purchase another domain? We can simply create a DNS alias that points to whatever we want....... we could just create `av-embed.api.bbci.co.uk` for instance.

We would need a cert, but that could just point to a CF distribution or something similar.

**Comment by matthew1000** [comment]
Oh yeah, definitely create a subdomain using Route 53... no need for an entirely new domain. I propose that the origin be Akamai, which OTG manage. So we can ask OTG to handle the cert.

The next question is whether we make a specific subdomain for this use-case (e.g. `av-embed.`) or a more generic one (e.g. `polling.` or `cdn.`).

I like the idea of a generic one (e.g. `cdn.api.bbci.co.uk`) that routes to Mozart/Belfrage via Akamai, for three reasons:

* It can be used in all kinds of ways... embeds, dynamic content, APIs, polling... (This is what we found with Forge and its generic `polling.bbc.co.uk`).
* It's fewer domains for clients to look up DNS
* It's fewer Akamai configurations and certs for OTG to maintain.

But I'm not super-fussed, so if you'd rather just create your own domain, that's cool with me too.

**Comment by pjlee11** [comment]
@hindsc52 @jamesdonoh @dgm34 any preference on the name of the domain? Any opinions regarding the routing logic?

@matthew1000 do you know if anyone in Web Core has any strong opinions regarding the naming of domains? 

I'm thinking something like `cdn.api.bbci.co.uk` and us routing to Mozart via a subdomain.

**Comment by dgm34** [comment]
I don't feel strongly about this and  am perfectly happy with `cdn.api.bbci.co.uk`.

**Comment by matthew1000** [comment]
I've updated the Confluence page to say `cdn.`
JohnathanI (Web Core architect) is happy with `cdn.`
Anyone talked to Mozart/Belfrage team about this?

**Comment by jamesdonoh** [comment]
In general I agree with using a hostname that's already being set up/used for Web Core as we're trying to reduce DNS lookups anyway (with things like the move to `static.files.bbci.co.uk`).

That's assuming a) it's easy enough to partition off traffic to different back-ends and b) we expect everything served off it to have the same caching and availability requirements. The Confluence page describes it as being for _Data (GraphQL or JSON)_ whereas the WS media embeds are HTML documents. Do we think a Web Core sport scores GraphQL response would expect the same freshness as a piece of WS media content? I think probably yes but please disagree if you don't.

Also I have no opinion about the actual name, `cdn.api` seems good as any.

**Comment by dgm34** [comment]
_"Anyone talked to Mozart/Belfrage team about this?"_

I have, briefly, and @ettomatic will comment on the proposals soon.

**Comment by ettomatic** [comment]
Thanks for involving us in the discussion!

Happy with `cdn.api.bbci.co.uk` pointing to `www.mozart-routing.api.bbci.co.uk`.

If we want to isolate potentially high-level traffic, we could also consider pointing `cdn` to `eu-west2` Mozart as we already do for `polling.bbc.co.uk/news/breaking-news`

**Comment by pjlee11** [comment]
I've created `TA-1586` in Jira to handle this.

**Comment by ryanmccombe** [comment]
**Note: AMP tests have been disabled due to this bug causing the cypress tests to fail**

These PRs should be reverted as part of closing this issue:

- ~https://github.com/bbc/simorgh/pull/4591~
- ~https://github.com/bbc/simorgh/pull/4599~

**Comment by jamesdonoh** [comment]
The above PRs linked by Ryan were already reverted in https://github.com/bbc/simorgh/pull/4639 - the exclusion added as part of this PR should be removed instead.

**Comment by ryanmccombe** [comment]
Related: https://github.com/bbc/simorgh/issues/4873

Should be fixed by: https://github.com/bbc/simorgh/pull/4874

**Comment by 12** [comment]
Our solution has been to serve media over the `polling.bbc.co.uk` domain. 

This work is now on test, and is proving to work for both canonical and amp. We will be going live tomorrow morning with the media player.
